packages/@aws-cdk/aws-servicediscovery/lib/service.ts

```diff
@@ -342,6 +377,10 @@ function renderDnsRecords(dnsRecordType: DnsRecordType, dnsTtl: Duration = Durat
   }
 }
 
+export function defaultDiscoveryType(namespace : INamespace): DiscoveryType {
```

Comment:
One last change. We shouldn't export this internal helper function. If it can't be a private class member, than the best way to do this is adding a `lib/private/utils.ts` file that is not exported from `index.ts`

Reply:
I think ideally it would be a function on the namespace; however I think that constitutes a breaking change as it would be add a required parameter to the interface. 
I have moved it into a utils class